src/planner/operator/logical_dummy_scan.cpp

```diff
@@ -3,12 +3,15 @@
 namespace duckdb {
 
 void LogicalDummyScan::Serialize(FieldWriter &writer) const {
-	throw NotImplementedException(LogicalOperatorToString(type));
+	writer.WriteField(table_index);
 }
 
 unique_ptr<LogicalOperator> LogicalDummyScan::Deserialize(ClientContext &context, LogicalOperatorType type,
                                                           FieldReader &reader) {
-	throw NotImplementedException(LogicalOperatorToString(type));
+	auto table_index = reader.ReadRequired<idx_t>();
+
+	auto result = make_unique<LogicalDummyScan>(table_index);
+	return result;
```

Comment:
Here (and in a few other instances) we could avoid creating the intermediate result:
```suggestion
	return make_unique<LogicalDummyScan>(table_index);
```
(just for readability: the compiler would have optimized that out anyway)

Reply:
Thanks, addressed this in this commit: https://github.com/hannes/duckdb/pull/4/commits/1290a67abab74a896a8281d4acd098a2bb86b97a